### PR TITLE
Manually implement equals check for candidate sets

### PR DESCRIPTION
**Pull request by SebastianSchmidl**

### CODE REVIEW

File: distod/src/main/scala/com/github/codelionx/distod/types/CandidateSet.scala [hunk 1]
```diff
@@ -165,4 +165,23 @@ class CandidateSet(private val _underlying: BitSet, private val _size: Int)
...
+      (this eq s) || s.size == this.size && check(s)
+    case _ => false
+  }
+
+  @inline private final def check(cs: CandidateSet): Boolean = {
+    val set1 = this._underlying.toBitMask
+    val set2 = cs._underlying.toBitMask
+    val len = set1.length max set2.length
+    var idx = 0
+    while (idx < len) {
```
Comment (CodeLionX) [inline distod/src/main/scala/com/github/codelionx/distod/types/CandidateSet.scala]: Codacy Issue found: [Avoid using while loops](https://app.codacy.com/manual/CodeLionX/distod/pullRequest?prid=5016225)

File: distod/src/main/scala/com/github/codelionx/distod/types/CandidateSet.scala [hunk 2]
```diff
@@ -165,4 +165,23 @@ class CandidateSet(private val _underlying: BitSet, private val _size: Int)
...
+      if(set1(idx) != set2(idx))
+        return false
```
Comment (CodeLionX) [inline distod/src/main/scala/com/github/codelionx/distod/types/CandidateSet.scala]: Codacy Issue found: [Avoid using return](https://app.codacy.com/manual/CodeLionX/distod/pullRequest?prid=5016225)

File: distod/src/main/scala/com/github/codelionx/distod/types/CandidateSet.scala [hunk 3]
```diff
@@ -165,4 +165,23 @@ class CandidateSet(private val _underlying: BitSet, private val _size: Int)
...
+    case s: CandidateSet =>
+      (this eq s) || s.size == this.size && check(s)
+    case _ => false
+  }
+
+  @inline private final def check(cs: CandidateSet): Boolean = {
+    val set1 = this._underlying.toBitMask
+    val set2 = cs._underlying.toBitMask
+    val len = set1.length max set2.length
+    var idx = 0
```
Comment (CodeLionX) [inline distod/src/main/scala/com/github/codelionx/distod/types/CandidateSet.scala]: Codacy Issue found: [Avoid mutable local variables](https://app.codacy.com/manual/CodeLionX/distod/pullRequest?prid=5016225)

File: distod/src/main/scala/com/github/codelionx/distod/types/CandidateSet.scala [hunk 4]
```diff
@@ -165,4 +165,23 @@ class CandidateSet(private val _underlying: BitSet, private val _size: Int)
...
+    var equal = true
+    while (equal && idx < len) {
```
Comment (SebastianSchmidl) [inline distod/src/main/scala/com/github/codelionx/distod/types/CandidateSet.scala:181]: Codacy Issue found: [Avoid using while loops](https://app.codacy.com/manual/CodeLionX/distod/pullRequest?prid=5016225)

File: distod/src/main/scala/com/github/codelionx/distod/types/CandidateSet.scala (hunk 3 above)
Comment (SebastianSchmidl) [inline distod/src/main/scala/com/github/codelionx/distod/types/CandidateSet.scala:179]: Codacy Issue found: [Avoid mutable local variables](https://app.codacy.com/manual/CodeLionX/distod/pullRequest?prid=5016225)

File: distod/src/main/scala/com/github/codelionx/distod/types/CandidateSet.scala [hunk 5]
```diff
@@ -165,4 +165,23 @@ class CandidateSet(private val _underlying: BitSet, private val _size: Int)
...
+      (this eq s) || s.size == this.size && check(s)
+    case _ => false
+  }
+
+  @inline private final def check(cs: CandidateSet): Boolean = {
+    val set1 = this._underlying.toBitMask
+    val set2 = cs._underlying.toBitMask
+    val len = set1.length max set2.length
+    var idx = 0
+    var equal = true
```
Comment (SebastianSchmidl) [inline distod/src/main/scala/com/github/codelionx/distod/types/CandidateSet.scala:180]: Codacy Issue found: [Avoid mutable local variables](https://app.codacy.com/manual/CodeLionX/distod/pullRequest?prid=5016225)

**Pull request merged**